Llk/TreeNode.php

```diff
@@ -176,6 +176,19 @@ public function getValueToken()
     }
 
     /**
+     * Get token offset.
+     *
+     * @return int
+     */
+    public function getOffset()
+    {
+        return
+            isset($this->_value['offset'])
```

Comment:
Why checking if the `offset` key exists?

Reply:
Token is not an immutable and $this->_value can be `null` for Rules. Only tokens contain the offset value:

```
$node->getValueValue(); // null
$node->getOffset(); // 0

$token->getValueValue(); // "Hello"
$token->getOffset(); // 42
```

Reply:
Didn't check the other methods, sorry. Make sense!

Reply:
Hmmm... Is not it worth it to make a `null` by default?

Because with `0` there can be problems when it is not clear whether this is a valid offset value or simply there is no data.

Reply:
An offset is an integer. 0 is an integer. Everything is OK here for me :-). If no value, then we strictly check against `null`.